App/src/App.tsx

```diff
@@ -7,25 +7,37 @@ import Home from "./pages/Home";
 import NotFound from "./pages/NotFound";
 import Login from "./pages/Login";
 import { User } from "./services/user";
-import { createContext, useEffect, useState } from "react";
+import { createContext, useState } from "react";
 import Register from "./pages/Register";
 import { decodeTokenToUser } from "./services/auth";
 import { BaseRoutes } from "./constants";
+import axios from "axios";
+import Dossiers from "./pages/dossier/Dossiers";
 
 export const UserContext = createContext<User | null>(null);
 
 export function App() {
-    const [user, setUser] = useState<User | null>(null);
+    const [user, setUser] = useState<User | null>(initializeUser());
+    const [isLoggedIn, setIsLoggedIn] = useState<boolean>(user != null ? true : false);
+    const navigate = useNavigate();
 
-    useEffect(() => {
-        // Check for the token in localStorage
+    // Check for the token in localStorage.
+    function initializeUser() {
```

Comment:
From curiosity, any reason you took this out of useEffect?

Reply:
because the initialization was being done asynchronously. I wanted the initialization of the token to happen the first thing otherwise some requests were getting unauthorized access

Reply:
It smells of a state management issue then. I think typically this sort of logic would all be in the user's useContext hook, and this hook would provide a getUser function (rather than directly passing the user object), which, when called, would do any checks like this. It could also refresh the token as needed. If this works though, no need to change it for now.